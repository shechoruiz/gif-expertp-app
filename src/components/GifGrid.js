import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=Goku&limit=10&api_key=kx88TEZM2i1M6R4HS35zxEarXqOLCjz1";
    const res = await fetch(url);
    const { data } = await res.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    console.log(gifs);
    setImages(gifs);
  };

  return (
    <div>
      <h3>{category}</h3>
      {images.map((img) => (
        <GifGridItem key={img.id} {...img} />
      ))}
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string,
};

export default GifGrid;
