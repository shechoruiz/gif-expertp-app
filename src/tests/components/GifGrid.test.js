import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs"); // Con este mock, se finje la llamada a ese archivo y suponer/controlar la información que va a responder

describe("Pruebas sobre el componente GifGrid", () => {
  const category = "Goku";

  test("1. Generación del MatchSnapshot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("2. Debe mostrar iutems cuando se cargan imagenes con el useFetchgifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "123",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
