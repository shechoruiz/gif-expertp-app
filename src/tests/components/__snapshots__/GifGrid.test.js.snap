// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifGrid 1. Generaci√≥n del MatchSnapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre el componente GifGrid 2. Debe mostrar iutems cuando se cargan imagenes con el useFetchgifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
